packages/kauri-wallet/src/app/wallet/wallet.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component } from '@angular/core';
+import { ApiService } from '../services/api/api.service';
+import { ApiModel } from '../services/api/api.model';
+import { DataService } from '../services/data/data.service';
+
+@Component({
+  selector: 'app-api-sample',
+  templateUrl: './wallet.component.html',
+  styleUrls: ['./wallet.component.scss']
+})
+export class WalletComponent {
+
+  constructor(
+    private _apiService: ApiService,
+    private _dataService: DataService
+  ) { }
+
+  onGetTransactions(currency: string, addresses: string[]) {
+
+    const apiModel: ApiModel = {} as ApiModel;
```

Comment:
I thought the `as` syntax was for overwriting the original type.
E.g. [Type Assertions](https://basarat.gitbooks.io/typescript/docs/types/type-assertion.html)

However, you seem to be doing it everywhere. 

Would just `const apiModel: ApiModel = {} as ApiModel;` work?

Or

```js
const transactionData: ApiModel = {
  currency,
  addresses,
};
```

Reply:
To take it a step further, does this work?

`wallet.component.html`
```js
  onGetTransactions(transactionsReq:  ApiModel) {
    this._apiService.getRawTransactions(transactionsReq);
  }
```

`wallet.component.html`

```html
  <button mat-raised-button (click)="onGetTransactions({currency.value,[addressOne.value, addressTwo.value]})">Get Transactions</button>
```

I am not sure if/how angular/typescript handle type errors in a template?

Reply:
As discussed we should do it this way.
`const apiModel: ApiModel = {} as ApiModel;`

but this is also valid `const apiModel = {} as ApiModel;`